soc/riscv/riscv-privilege/andes_v5/ae350/linker.ld

```diff
@@ -87,9 +87,7 @@ MEMORY
 #endif
     RAM (rwx) : ORIGIN = RAM_BASE, LENGTH = RAM_SIZE
 
-    /* Data & Instruction Tightly Coupled Memory */
-    LINKER_DT_REGION_FROM_NODE(DT_CHOSEN(zephyr_itcm), rw)
-    LINKER_DT_REGION_FROM_NODE(DT_CHOSEN(zephyr_dtcm), rw)
+    LINKER_DT_REGIONS()
```

Comment:
When testing with `arty_a7_arm_designstart_m3` this code results in `DTCM` and `ITCM` being defined on the same line like this:

```
MEMORY
    {
    FLASH (rx) : ORIGIN = (0x0 + 0x0), LENGTH = (32*1K - 0x0)
    SRAM (wx) : ORIGIN = 0x20000000, LENGTH = (32 * 1K)
    "ITCM" : ORIGIN = 0, LENGTH = 32768 "DTCM" : ORIGIN = 536870912, LENGTH = 32768
    IDT_LIST (wx) : ORIGIN = 0xFFFFF7FF, LENGTH = 2K
    }
```

it's not invalid code, but would be nice if each could have their own line.

This looks very nice when using `-DCONFIG_CMAKE_LINKER_GENERATOR=y` :wink:

Reply:
I agree, not easy at all and I can't think of a solution, just wanted to mention it.